File: Objects/Word.cs
using System.Collections.Generic;

namespace Crossword.Objects;

public class Word
{
    public List<Cell> Cells { get; } = new();
    public List<Cell> ConnectionCells { get; } = new();
    public List<Word> ConnectionWords { get; } = new();
    public bool Full { get; set; }
    public string WordString { get; set; } = "";
    public bool Right { get; set; }
    public List<Dictionary> FullDictionaries { get; } = new();
    public bool Fix { get; set; } = false;
}

File: crossword_python/models/Word.py
from dataclasses import dataclass, field
from .Cell import Cell
from .Dictionary import Dictionary

@dataclass
class Word:
    full: bool = False
    word_string: str = ""
    right: bool = False
    fix: bool = False
    cells: list[Cell] = field(default_factory=lambda: [ ])
    connection_cells: list[Cell] = field(default_factory=lambda: [ ])
    connection_words: list['Word'] = field(default_factory=lambda: [ ])
    full_dictionaries: list[Dictionary] = field(default_factory=lambda: [ ])